AK/String.cpp

```diff
@@ -122,12 +122,30 @@ ByteBuffer String::to_byte_buffer() const
     return ByteBuffer::copy(reinterpret_cast<const byte*>(characters()), length());
 }
 
-// FIXME: Duh.
 int String::to_int(bool& ok) const
 {
-    unsigned value = to_uint(ok);
-    ASSERT(ok);
-    return (int)value;
+  bool negative = false;
+  int value = 0;
+  ssize_t i = 0;
+  if (characters()[0] == '-') {
```

Comment:
characters() may return null, so we must check that before using it like this.
The function could just start with
```
if (is_null()) {
    ok = false;
    return 0;
}
```